src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -101,8 +101,49 @@ export class WorkspacesUsecases {
     };
   }
 
-  async setupWorkspace(user: User, createWorkSpaceDto: CreateWorkSpaceDto) {
-    //return this.teamRepository.createTeam(newTeam);
+  async setupWorkspace(
+    user: User,
+    workspaceId: WorkspaceAttributes['id'],
+    setupWorkspaceDto: SetupWorkspaceDto,
+  ) {
+    const workspace = await this.workspaceRepository.findOne({
+      id: workspaceId,
+      ownerId: user.uuid,
+      setupCompleted: false,
+    });
+
+    if (!workspace) {
+      throw new NotFoundException('There is no workspace to be setup');
+    }
+
+    const workspaceUser = WorkspaceUser.build({
+      id: v4(),
+      workspaceId: workspaceId,
+      memberId: user.uuid,
+      spaceLimit: BigInt(0),
+      driveUsage: BigInt(0),
+      backupsUsage: BigInt(0),
+      key: setupWorkspaceDto.encryptedMnemonic,
+      deactivated: false,
+      createdAt: new Date(),
+      updatedAt: new Date(),
+    });
+
+    await this.workspaceRepository.addUserToWorkspace(workspaceUser);
```

Comment:
If this step fails, we may want to check which step is still failing, so in the case of a massive failure, we can just fix the error and users will only need to retry again. What do you think?